src/Analyzer.Utilities/JsonLineNumberResolver.cs

```diff
@@ -37,8 +39,23 @@ public JsonLineNumberResolver(TemplateContext templateContext)
         /// or 1 if it can't be determined.</returns>
         public int ResolveLineNumber(string pathInExpandedTemplate)
```

Comment:
Remember to add tests

Reply:
The tests I have in TemplateAnalyzerTests.cs check for specific line numbers, which is part of what I was using to debug the resolveLineNumber function. Would that be sufficient by any chance?

Reply:
Added some tests and modified the function to make it fully backward operating